addons/flags/XEH_postInit.sqf

```diff
@@ -0,0 +1,39 @@
+#include "script_component.hpp"
+
+if (!hasinterface) exitWith {};
+
+GVAR(isPlacing) = PLACE_CANCEL;
+["ace_interactMenuOpened", {GVAR(isPlacing) = PLACE_CANCEL;}] call CBA_fnc_addEventHandler;
+
+[QGVAR(flagPlaced), {
+    params ["_unit", "_item", "_flag"];
+
+    (GVAR(flagItemCache) get _item) params ["_flagName"];
+
+    private _pickupFlag = [
+        QGVAR(pickup),
+        format [LLSTRING(Pickup), _flagName],
+        QPATHTOF(data\icons\place\white_pickup_icon.paa),
+        {
+            params ["_flag", "_unit", "_item"];
+            [_unit, _item, _flag] call FUNC(pickupFlag);
+        },
+        {true},
+        {},
+        _item,
+        [0, -0.45, 0.75],
+        2
+    ] call ace_interact_menu_fnc_createAction;
+    [_flag, 0, [], _pickupFlag] call ace_interact_menu_fnc_addActionToObject;
+}] call CBA_fnc_addEventHandler;
+
+private _flagItems = (call (uiNamespace getVariable [QGVAR(allFlagItems), {[]}])) apply {configFile >> "CfgWeapons" >> _x};
```

Comment:
This could be sped up a bit by storing configFile >> "CfgWeapons" in a local variable, instead of re-resolving it every iteration.
But not much